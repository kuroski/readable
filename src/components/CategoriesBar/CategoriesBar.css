.categories-bar {
  --bg-color: #f60;
  --link-color: hsla(0, 0%, 100%, 0.8);
  --link-hover-color: #fff;
}

.categories-bar {
  background-color: var(--bg-color);
  padding: 1rem;
  display: flex;
}

.categories-bar__link {
  color: var(--link-color);
  transition: color 0.15s ease;
  vertical-align: middle;
  text-decoration: none;
  margin-left: 1rem;
  text-transform: capitalize;
}

.categories-bar__link:hover {
  color: var(--link-hover-color);
}

.categories-bar__link:first-child {
  margin-left: 0;
}
