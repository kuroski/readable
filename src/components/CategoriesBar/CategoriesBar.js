import React from 'react';
import createReactClass from 'create-react-class';
import Link from 'react-router-dom/Link';
import PropTypes from 'prop-types';
import './CategoriesBar.css';

const CategoriesBar = createReactClass({
  render: function() {
    return (
      <div>
        CategoriesBar -
        <ul>
          {this.props.categories.map(category => (
            <li key={category.name}>
              <Link to={category.path}>{category.name}</Link>
            </li>
          ))}
        </ul>
      </div>
    );
  }
});

CategoriesBar.propTypes = {
  categories: PropTypes.array.isRequired
};

CategoriesBar.defaultProps = {
  categories: []
};

export default CategoriesBar;
