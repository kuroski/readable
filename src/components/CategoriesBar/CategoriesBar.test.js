import React from 'react';
import renderer from 'react-test-renderer';
import Link from 'react-router-dom/Link';
import shallow from 'enzyme/shallow';
import CategoriesBar from './CategoriesBar';
import { categories } from '../../common/testData';

describe('CategoriesBar component', () => {
  let props;

  const build = () => {
    return shallow(<CategoriesBar {...props} />);
  };

  beforeEach(() => {
    props = {
      categories: categories
    };
  });

  it('renders CategoriesBar component correctly', () => {
    const wrapper = build();
    const CategoriesBarTree = renderer.create(wrapper[0]).toJSON();
    expect(CategoriesBarTree).toMatchSnapshot();
  });

  it('must show all categories', () => {
    const wrapper = build();
    props.categories.forEach(category =>
      expect(wrapper.contains(category.name)).toBe(true)
    );
  });

  it('must show categories links', () => {
    const wrapper = build();
    props.categories.forEach(category => {
      const link = <Link to={category.path}>{category.name}</Link>;
      expect(wrapper.contains(link)).toBe(true);
    });
  });
});
