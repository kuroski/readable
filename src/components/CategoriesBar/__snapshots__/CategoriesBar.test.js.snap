// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesBar component renders CategoriesBar component correctly 1`] = `null`;
